soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h7a3xx

```diff
@@ -3,13 +3,26 @@
 # Copyright (c) 2021 Electrolance Solutions
 # SPDX-License-Identifier: Apache-2.0
 
+if SOC_STM32H7A3XX || SOC_STM32H7A3XXQ
+
+config NUM_IRQS
+	default 155
+
+endif #SOC_STM32H7A3XX || SOC_STM32H7A3XXQ
+
 if SOC_STM32H7A3XX
 
 config SOC
-	default "stm32h7a3xxQ" if POWER_SUPPLY_SMPS
-	default "stm32h7a3xx" if !POWER_SUPPLY_SMPS
+	default "stm32h7a3xx"
```

Comment:
In this file, the only change required is this one:
```
	default "stm32h7a3xxQ" if HAS_POWER_SUPPLY_SMPS
	default "stm32h7a3xx" if !HAS_POWER_SUPPLY_SMPS
```

This is correct if `HAS_POWER_SUPPLY_SMPS` isn't user selectable but only indirectly selected by the use of `SOC_STM32H7A3XXQ` in board definition

Reply:
Let's keep this part the way it is. This is fine

Reply:
Do you mean fallback to condition upon `POWER_SUPPLY_SMPS`?